Public/Src/Cache/ContentStore/Library/Sessions/ServiceClientContentSession.cs

```diff
@@ -38,65 +38,47 @@ public class ServiceClientContentSession : ReadOnlyServiceClientContentSession,
         }
 
         /// <inheritdoc />
-        protected override Task<PutResult> PutStreamCoreAsync(
+        protected override async Task<PutResult> PutStreamCoreAsync(
             OperationContext operationContext, HashType hashType, Stream stream, UrgencyHint urgencyHint, Counter retryCounter)
         {
             // We need a seekable stream, that can give its length. If the input stream is seekable, we can use it directly.
             // Otherwise, we need to create a temp file for this purpose.
-            Stream putStream;
-            long position;
-            Stream disposableStream;
-
-            if (stream.CanSeek)
-            {
-                putStream = stream;
-                position = stream.Position;
-                disposableStream = null;
-            }
-            else
+            var putStream = stream;
+            Stream disposableStream = null;
+            if (!stream.CanSeek)
             {
                 putStream = TempFileStreamFactory.Create(operationContext, stream);
-                position = 0;
                 disposableStream = putStream;
             }
 
             using (disposableStream)

```

Comment:
disposableStream is still not used... Is it ok?

Reply:
Ignore this. It is fine. putStream is used. So we're good.

---
In reply to: [292134755](https://github.com/microsoft/BuildXL/pull/431#discussion_r292134755) [](ancestors = 292134755)